omise/__init__.py

```diff
@@ -811,6 +847,80 @@ def schedule(self):
         return schedules
 
 
+class LazyCollection(object):
+    def __init__(self, collection_path):
+        self.collection_path = collection_path
+        self._exhausted = False
+
+    def __len__(self):
+        obj = self._fetch_objects(limit=1, offset=0)
+
+        return obj["total"]
+
+    def __iter__(self):
+        self._reset_listing()
+        return self
+
+    def __next__(self):
+        if self.listing is None or self._list_index + 1 > len(self.listing):
+            if self._exhausted:
+                raise StopIteration
+            self._next_batch()
+
+        self._list_index += 1
+        return _as_object(self.listing[self._list_index - 1])
+
+    def next(self):
+        return self.__next__()
+
+    def _next_batch(self, limit=100):
+        offset = self._list_index
+
+        obj = self._fetch_objects(limit=limit, offset=offset)
+        data = obj["data"]
+
+        if len(data) > 0:
+            self._add_to_listing(data)
+
+            if len(data) < limit:
+                self._exhausted = True
+        else:
+            raise StopIteration
+
+    def _reset_listing(self):
+        self._list_index = 0
+        self.listing = None
+        self._exhausted = False
+
+    def _add_to_listing(self, data):
+        if self.listing:
+            self.listing.append(data)
+        else:
+            self.listing = data
+
+    def _fetch_objects(self, **kwargs):
+        order = kwargs.pop('order', None)
+        return Request(api_secret, api_main, api_version).send(
+            'get',
+            self.collection_path,
+            payload={
+                'limit': kwargs['limit'],
+                'offset': kwargs['offset'],
+                'order': order
```

Comment:
Hmm, can it be like this
`'order': kwargs.pop('order', None)` 
because I don't see any other `order` var usage except this place 😄